### PR TITLE
feat: Integrate the bugfixes of Redis 7.2.5

### PR DESCRIPTION
**Pull request by SoulPancake**
fixes 
- #561

### CODE REVIEW

**Comment by SoulPancake** [comment]
Can you take a look here @madolson 
Since unstable already had all these changes, I decided to accept those during the conflicts while basing of my branch from 7.2

**Comment by hwware** [comment]
@SoulPancake DCO missed, pls sign off

**Comment by SoulPancake** [comment]
My bad, done @hwware

**Comment by PingXie** [comment]
The size of this PR doesn't look right and cluster_legacy.c doesn't exist on 7.2. Did you bring over the unstable changes?

**Comment by SoulPancake** [comment]
Yes I think so
So when I was cherry picking b3aaa0a1362d229ba1ecd44629655f76c77304ec
I accepted unstable changes, as I could see the entire changes there from the bugfix commits
@PingXie 

image

**Comment by hwware** [comment]
I think you need to update some test cases executable file name,  such as valkey-check-aof etc

File: tests/integration/aof.tcl [hunk 1]
```diff
@@ -155,24 +119,24 @@ tags {"aof external:skip"} {
         }
 
         catch {
-            exec src/valkey-check-aof $aof_manifest_file
+            exec src/redis-check-aof $aof_manifest_file
```
Comment (hwware) [inline tests/integration/aof.tcl]: update to valkey-check-aof, otherwise test case fails

**Pull request closed without merging**